src/sample_inverted_index.cpp

```diff
@@ -0,0 +1,93 @@
+#include <cmath>
+#include <fstream>
+#include <random>
+#include <string>
+
+#include "CLI/CLI.hpp"
+#include "binary_freq_collection.hpp"
+#include "invert.hpp"
+#include "util/progress.hpp"
+
+using namespace pisa;
+
+template <typename SampleFn>
+void sample_index(std::string const &input_basename,
+                  std::string const &output_basename,
+                  SampleFn &&sample_fn)
+{
+
+    binary_freq_collection input(input_basename.c_str());
+
+    boost::filesystem::copy_file(fmt::format("{}.sizes", input_basename),
+                                 fmt::format("{}.sizes", output_basename),
+                                 boost::filesystem::copy_option::overwrite_if_exists);
+
+    std::ofstream dos(output_basename + ".docs");
+    std::ofstream fos(output_basename + ".freqs");
+
+    auto document_count = static_cast<uint32_t>(input.num_docs());
+    write_sequence(dos, gsl::make_span<uint32_t const>(&document_count, 1));
+
+    {
+        pisa::progress progress("Sampling inverted index", input.size());
+        for (auto const &plist : input) {
+            std::vector<uint32_t> docs(plist.docs.begin(), plist.docs.end());
+            std::vector<uint32_t> freqs(plist.freqs.begin(), plist.freqs.end());
```

Comment:
Are sequences in the collection not random access? If not, they should (open an issue?). If they are, we can just pass them instead of vectors, right?

Reply:
I will make it random accessible

Reply:
Awesome :+1: